test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsights/ApplicationInsightsEndToEndTests.cs

```diff
@@ -447,6 +455,236 @@ public async Task ApplicationInsights_HttpRequestTrackingByWebJobs(string testNa
+                Startup.Host = host;
+                await host.StartAsync();
+
+                var loggerProvider = host.Services.GetServices<ILoggerProvider>().OfType<ApplicationInsightsLoggerProvider>().Single();
+                var logger = loggerProvider.CreateLogger(LogCategories.Results);
+
+                string parentId = "|legacyRoot.1.2.3.";
+
+                // simulate functions behavior to set request on the scope
+                using (var _ = logger.BeginScope(new Dictionary<string, object>
+                {
+                    ["Links"] = new[] { new Activity("event hub process").SetParentId(parentId) }
+                }))
+                {
+                    MethodInfo methodInfo = GetType().GetMethod(testName, BindingFlags.Public | BindingFlags.Static);
+                    await host.GetJobHost().CallAsync(methodInfo, new { input = "function input" });
+                }
+
+                await host.StopAsync();
+
+                // Validate the request
+                RequestTelemetry functionRequest = _channel.Telemetries.OfType<RequestTelemetry>().SingleOrDefault();
+
+                ValidateRequest(
+                    functionRequest,
+                    testName,
+                    testName,
+                    null,
+                    null,
+                    true,
+                    "0",
+                    null,
+                    parentId);
+
+                // Make sure operation ids match
+                var traces = _channel.Telemetries.OfType<TraceTelemetry>().Where(
+                    t => t.Context.Operation.Id == functionRequest.Context.Operation.Id);
+                Assert.Equal(4, traces.Count());
+                Assert.False(functionRequest.Properties.ContainsKey("_MS.links"));
+
+                Assert.True(functionRequest.Properties.TryGetValue("ai_legacyRootId", out var legacyRoot));
+                Assert.Equal("legacyRoot", legacyRoot);
+                Assert.False(functionRequest.Properties.ContainsKey("Links"));
+            }
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task ApplicationInsights_EventHubTrackingByWebJobs_SingleLink(bool sampledIn)
+        {
+            string testName = nameof(TestApplicationInsightsInformation);
+
+            using (IHost host = ConfigureHost())
+            {
+                Startup.Host = host;
+                await host.StartAsync();
+
+                var loggerProvider = host.Services.GetServices<ILoggerProvider>().OfType<ApplicationInsightsLoggerProvider>().Single();
+                var logger = loggerProvider.CreateLogger(LogCategories.Results);
+
+                string traceId = ActivityTraceId.CreateRandom().ToHexString();
+                string parentSpanId = ActivitySpanId.CreateRandom().ToHexString();
+                string traceFlags = sampledIn ? "01" : "00";
+                string parentId = $"00-{traceId}-{parentSpanId}-{traceFlags}";
```

Comment:
Oh, so sampling is part of the trace context id? cool :-)

edit -- looks like it: https://www.w3.org/TR/trace-context/#trace-flags